src/Type/ClosureType.php

```diff
@@ -143,6 +143,11 @@ public function isPure(): TrinaryLogic
 		return $certainCount > 0 ? TrinaryLogic::createNo() : TrinaryLogic::createMaybe();
 	}
 
+	public function isDeprecated(): TrinaryLogic
+	{
+		return TrinaryLogic::createNo();
```

Comment:
This is not the case. First-class callables like `Foo::method(...)` are resolved to ClosureType so the deprecated-ness should be carried over.

Reply:
I am not sure how to deal with it. Do you think the class ClosureType itself should have a member "isDeprecated" which will be in the constructor and resolved during the creation of instance of ClosureType itself?